### PR TITLE
Implement `repository_set_namespace` function

### PR DESCRIPTION
**Pull request by gilbertw1**
This pull request implements the `repository_set_namespace` function.

I plan on implementing more functions, I'd just like to get this initial one in to verify that I've done everything correctly.

Thanks!

### CODE REVIEW

File: src/egit-repository.c [hunk 1]
```diff
@@ -306,6 +306,26 @@ emacs_value egit_repository_set_ident(
...
+{
+    EGIT_ASSERT_REPOSITORY(_repo);
+    EGIT_ASSERT_STRING(_nmspace);
+
+    git_repository *repo = EGIT_EXTRACT(_repo);
+    int retval;
+        {
+            char *nmspace = EGIT_EXTRACT_STRING(_nmspace);
+            retval = git_repository_set_namespace(repo, nmspace);
+            EGIT_FREE(nmspace);
```
Comment (TheBB) [inline src/egit-repository.c]: It's fine to use `free` here since `nmspace` shouldn't be null.

**Comment by gilbertw1** [comment]
Sorry, looks like my config had some wonky formatting setting for C and I totally missed it. I've made the requested updates.

Regarding a test for this function, it looks like namespaces are only intended to be used when working with remote repositories, so I'm not sure how feasible a proper test will be to write. Either way, I've put the stub in there for now.

**Comment by asavonic** [comment]
> Regarding a test for this function, it looks like namespaces are only intended to be used when working with remote repositories, so I'm not sure how feasible a proper test will be to write. Either way, I've put the stub in there for now.

So the main problem is that you don't have a 'remote' repository?
You can actually create two local repositories and add one of them as a remote by specifying a full path:

    (with-temp-dir first-repo
      (init)
      (with-temp-dir second-repo
        (init)
        (run "git" "remote" "add" second-repo)))

**Comment by TheBB** [comment]
Thanks!

**Pull request merged**